shell/platform/darwin/ios/rendering_api_selection.mm

```diff
@@ -21,26 +22,40 @@ bool ShouldUseMetalRenderer() {
   // past iOS 10.0. The processor was selected as it is the first version at which Metal was
   // supported. The iOS version floor was selected due to the availability of features used by Skia.
   bool ios_version_supports_metal = false;
-  if (@available(iOS 10.0, *)) {
+  if (@available(iOS METAL_IOS_VERSION_BASELINE, *)) {
     auto device = MTLCreateSystemDefaultDevice();
     ios_version_supports_metal = [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily1_v3];
   }
   return ios_version_supports_metal;
 }
 #endif  // FLUTTER_SHELL_ENABLE_METAL
 
-IOSRenderingAPI GetRenderingAPIForProcess() {
-#if TARGET_IPHONE_SIMULATOR
-  return IOSRenderingAPI::kSoftware;
-#endif  // TARGET_IPHONE_SIMULATOR
+IOSRenderingAPI GetRenderingAPIForProcess(bool force_software) {
+#if TARGET_OS_SIMULATOR
+  if (force_software) {
+    return IOSRenderingAPI::kSoftware;
+  }
+#elif !FLUTTER_RELEASE
```

Comment:
Good call on the log. Why `!FLUTTER_RELEASE` though? This is still a warning log that will only show up when verbose logging is enabled. I'd just use `#else`.

Reply:
I figure avoid the overhead if you're on release, since you should've seen it already in debug/profile if you're trying. I guess it's probably not a ton of overhead to do the bool check though.. I'll change it to else.